homeassistant/components/smappee/__init__.py

```diff
@@ -26,6 +26,7 @@
             {
                 vol.Required(CONF_CLIENT_ID): cv.string,
                 vol.Required(CONF_CLIENT_SECRET): cv.string,
+                vol.Optional(CONF_PLATFORM, default="PRODUCTION"): cv.string,
```

Comment:
We should not add this to our YAML schema, as it violates architectural decision record 10:

<https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

> Changes to existing YAML configuration for these same existing integrations, will no longer be accepted.

@MartinHjelmare Please advise.

Reply:
Discussed it shortly with @MartinHjelmare via chat.

Our current opinion is that this should not be added to the YAML configuration schema, for 2 reasons:

- It violates ADR-0010 as linked above
- It is not an option that should be available to the regular user (as it is merely meant for development).

The current suggestion is to move such logic into an environment variable. This way, a developer can still override/set it during development, without exposing the option in the configuration schema and thus not violating ADR-0010 as well.

Reply:
Ok